z3rro/src/prover.rs

```diff
@@ -94,6 +160,21 @@ impl<'ctx> Prover<'ctx> {
         if self.min_level_with_provables.is_none() {
             return ProveResult::Proof;
         }
+
+        let mut smtlib = self.get_smtlib();
```

Comment:
As a next step, add an `solver_type` attribute of a new enum type `SolverType` with variants Z3 and SWINE, and then choose here whether to invoke either Z3 or SWINE.